beacon-chain/rpc/beacon/validators.go

```diff
@@ -505,6 +509,48 @@ func (bs *Server) GetValidatorParticipation(
 	}, nil
 }
 
+// This appends non finalized block atts to state. To replay for a state, a node does not replay non canonical
+// nor finalized blocks. To count participation, this includes the attestations from the orphaned block to state.
+func (bs *Server) appendNonFinalizedBlockAttsToState(ctx context.Context, s *statetrie.BeaconState, e uint64) (*statetrie.BeaconState, error) {
+	start := helpers.StartSlot(e)
+	end := helpers.StartSlot(e + 1)
+	f := filters.NewFilter().SetStartSlot(start).SetEndSlot(end)
+	blks, err := bs.BeaconDB.Blocks(ctx, f)
+	if err != nil {
+		return nil, err
+	}
+	bRoots, err := bs.BeaconDB.BlockRoots(ctx, f)
+	if err != nil {
+		return nil, err
+	}
+	if len(blks) != len(bRoots) {
+		return nil, err
+	}
+	fbs := make([]*ethpb.SignedBeaconBlock, 0, len(blks))
+	for i := len(blks) - 1; i >= 0; i-- {
```

Comment:
what is the point of this, this slice isn't used anywhere after

Reply:
Nice catch. Fixed this in the latest commit. Thanks!